kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/DataSkippingUtils.java

```diff
@@ -0,0 +1,290 @@
+            // turn means we were able to prove that `a` and/or `b` will not evaluate to TRUE for any row
+            // of the file. If that is the case, then `AND(a, b)` also cannot evaluate to TRUE for any row
+            // of the file, which proves we have a valid data skipping predicate.
+            //
+            // NOTE: AND is special -- we can safely skip the file if one leg does not evaluate to TRUE,
+            // even if we cannot construct a skipping filter for the other leg.
+            case "AND":
+                Optional<Predicate> e1Filter = constructDataFilters(
+                    asPredicate(getLeft(dataFilter)), schemaHelper);
+                Optional<Predicate> e2Filter = constructDataFilters(
+                    asPredicate(getRight(dataFilter)), schemaHelper);
+                if (e1Filter.isPresent() && e2Filter.isPresent()) {
+                    return Optional.of(new And(e1Filter.get(), e2Filter.get()));
+                } else if (e1Filter.isPresent()) {
+                    return e1Filter;
+                } else {
+                    return e2Filter; // possibly none
+                }
+
+            case "=": case "<": case "<=": case ">": case ">=":
+                Expression left = getLeft(dataFilter);
+                Expression right = getRight(dataFilter);
+
+                if (left instanceof Column && right instanceof Literal) {
+                    Column leftCol = (Column) left;
+                    Literal rightLit = (Literal) right;
+                    if (schemaHelper.isSkippingEligibleMinMaxColumn(leftCol) &&
+                        schemaHelper.isSkippingEligibleLiteral(rightLit)) {
+                        return Optional.of(constructComparatorDataFilters(
+                            dataFilter.getName(), leftCol, rightLit, schemaHelper));
+                    }
+                } else if (right instanceof Column && left instanceof Literal) {
+                    return constructDataFilters(reverseComparatorFilter(dataFilter), schemaHelper);
+                }
+
+            // TODO more expressions
+            default:
+                return Optional.empty();
+        }
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////
+    // Useful helper functions for dealing with expressions
+    //////////////////////////////////////////////////////////////////////////////////
+
+    /** Construct the skipping predicate for a given comparator */
+    private static Predicate constructComparatorDataFilters(
+            String comparator, Column leftCol, Literal rightLit, StatsSchemaHelper schemaHelper) {
+
+        switch (comparator.toUpperCase(Locale.ROOT)) {
+
+            // Match any file whose min/max range contains the requested point.
+            case "=":
+                return new And(
+                    new Predicate("<=", schemaHelper.getMinColumn(leftCol), rightLit),
```

Comment:
this is a bit hard to read, could you please add the simplified SQL expression as comment.